pandas/plotting/_matplotlib/timeseries.py

```diff
@@ -310,7 +310,7 @@ def maybe_convert_index(ax: Axes, data: NDFrameT) -> NDFrameT:
             if isinstance(data.index, ABCDatetimeIndex):
                 data = data.tz_localize(None).to_period(freq=freq_str)
             elif isinstance(data.index, ABCPeriodIndex):
-                data.index = data.index.asfreq(freq=freq_str)
+                data.index = data.index.asfreq(freq=freq_str, how = "S") #Lineplot alignes to the start of period  
```

Comment:
Could you please check `ruff`, `ruff-format` and `codespell` errors on `line 313`?
If you try following command: `pre-commit run --all-file`, it should help fix these errors.